drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -251,25 +293,94 @@ static int adc_smartbond_read_async(const struct device *dev,
 }
 #endif /* CONFIG_ADC_ASYNC */
 
+#ifdef CONFIG_PM_DEVICE
+static int gpadc_smartbond_resume(const struct device *dev)
+{
+	int ret;
+	struct adc_smartbond_data *data = dev->data;
+	const struct adc_smartbond_cfg *config = dev->config;
+
+	GPADC->GP_ADC_CTRL_REG = data->gpadc_ctrl_reg;
+	GPADC->GP_ADC_CTRL2_REG = data->gpadc_ctrl2_reg;
+
+	/*
+	 * Configure dt provided device signals when available.
+	 * pinctrl is optional so ENOENT is not setup failure.
+	 */
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0 && ret != -ENOENT) {
+		LOG_ERR("ADC pinctrl setup failed (%d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int gpadc_smartbond_suspend(const struct device *dev)
+{
+	int ret;
+	struct adc_smartbond_data *data = dev->data;
+	const struct adc_smartbond_cfg *config = dev->config;
+
+	data->gpadc_ctrl_reg = GPADC->GP_ADC_CTRL_REG;
+	data->gpadc_ctrl2_reg = GPADC->GP_ADC_CTRL2_REG;
+	/* Disable the GPADC LDO */
+	GPADC->GP_ADC_CTRL_REG = 0;
+
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
+	if (ret < 0) {
+		LOG_WRN("Failed to configure the GPADC pins to inactive state");
+	}
+
+	return ret;
+}
+
+static int gpadc_smartbond_pm_action(const struct device *dev,
+				   enum pm_device_action action)
+{
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		da1469x_pd_acquire(MCU_PD_DOMAIN_PER);
+		gpadc_smartbond_resume(dev);
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		gpadc_smartbond_suspend(dev);
+		da1469x_pd_release(MCU_PD_DOMAIN_PER);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+	return 0;
+}
+#endif /* CONFIG_PM_DEVICE */
+
 static int adc_smartbond_init(const struct device *dev)
 {
-	int err;
+	int ret;
 	struct adc_smartbond_data *data = dev->data;
 	const struct adc_smartbond_cfg *config = dev->config;
 
-	GPADC->GP_ADC_CTRL_REG = GPADC_GP_ADC_CTRL_REG_GP_ADC_EN_Msk;
-	GPADC->GP_ADC_CTRL2_REG = 0;
-	GPADC->GP_ADC_CTRL3_REG = 0x40;
-	GPADC->GP_ADC_CLEAR_INT_REG = 0x0;
+#ifdef CONFIG_PM_DEVICE_RUNTIME
+	/* Make sure device state is marked as suspended */
+	pm_device_init_suspended(dev);
+
+	ret = pm_device_runtime_enable(dev);
+
+#else
+	da1469x_pd_acquire(MCU_PD_DOMAIN_PER);
+	ret = gpadc_smartbond_resume(dev);
+#endif
 
+	GPADC->GP_ADC_CLEAR_INT_REG = 0x0;
 	/*
 	 * Configure dt provided device signals when available.
 	 * pinctrl is optional so ENOENT is not setup failure.
 	 */
-	err = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
-	if (err < 0 && err != -ENOENT) {
-		LOG_ERR("ADC pinctrl setup failed (%d)", err);
-		return err;
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
```

Comment:
This should be redundant as the default pins definition takes place in `gpadc_smartbond_resume`. Besides, if `ret` is negative and `PD_PER` is released, an assertion should be raised when in device runtime (`da1469x_pd_acquire` should not be called at this moment).